core/services/gateway/handlers/webapicapabilities/handler.go

```diff
@@ -176,12 +215,117 @@ func (h *handler) handleWebAPITriggerMessage(ctx context.Context, msg *api.Messa
 }
 
+//	body := api.MessageBody{
+//		MessageId: types.RandomID().String(),
+//		DonId:     h.connector.DonID(),
+//		Method:    webapicapabilities.MethodWebAPITriggerUpdateMetadata,
+//		Receiver:  gatewayIDStr,
+//		Payload:   payloadJSON,
+//	}
+func (h *handler) handleWebAPITriggerUpdateMetadata(ctx context.Context, msg *api.Message, nodeAddr string) error {
+	body := msg.Body
+	h.lggr.Debugw("handleWebAPITriggerUpdateMetadata", "body", body, "payload", string(body.Payload))
+
+	var payload map[string]string
+	err := json.Unmarshal(body.Payload, &payload)
+	donID := body.DonId
+	if err != nil {
+		// errors here:
+		// error decoding payload	{"version": "unset@unset",
+		// "err": "json: cannot unmarshal object into Go struct field Map.Underlying of type values.Value",
+		// "payload": "{\"Underlying\":{\"AllowedSenders\":{\"Underlying\":[{\"Underlying\":\"0x853d51d5d9935964267a5050aC53aa63ECA39bc5\"}]},\"AllowedTopics\":{\"Underlying\":[{\"Underlying\":\"daily_price_update\"},{\"Underlying\":\"ad_hoc_price_update\"}]},\"RateLimiter\":{\"Underlying\":{\"GlobalBurst\":{\"Underlying\":101},\"GlobalRPS\":{\"Underlying\":100},\"PerSenderBurst\":{\"Underlying\":103},\"PerSenderRPS\":{\"Underlying\":102}}},\"RequiredParams\":{\"Underlying\":[{\"Underlying\":\"bid\"},{\"Underlying\":\"ask\"}]}}}"}
+		h.lggr.Errorw("error decoding payload", "err", err, "payload", string(body.Payload))
+		// callbackCh <- handlers.UserCallbackPayload{Msg: msg, ErrCode: api.UserMessageParseError, ErrMsg: fmt.Sprintf("error decoding payload %s", err.Error())}
+		// close(callbackCh)
+		return err
+	}
+	triggersConfig := make(map[string]webapicap.TriggerConfig)
+	for triggerID, configPbString := range payload {
+		pbBytes, err := base64.StdEncoding.DecodeString(configPbString)
+		if err != nil {
+			h.lggr.Errorw("error decoding pb bytes", "err", err)
+			return err
+		}
+		pb := &pb.Map{}
+		proto.Unmarshal(pbBytes, pb)
+		vmap, err := values.FromMapValueProto(pb)
+		if err != nil {
+			h.lggr.Errorw("error FromMapValueProto", "err", err)
+			return err
+		}
+		if vmap == nil {
+			h.lggr.Errorw("error FromMapValueProto nil vmap")
+			return err
+		}
+		reqConfig, err := h.ValidateConfig(vmap)
+		if err != nil {
+			h.lggr.Errorw("error validating config", "err", err)
+			return err
+		}
+		triggersConfig[triggerID] = *reqConfig
+	}
+	h.triggersConfig.triggersConfigMap[donID] = TriggersConfig{lastUpdatedAt: time.Now(), triggersConfig: triggersConfig}
+
+	// h.updateTriggerConsensus()
+	return nil
+}
+
+func (h *handler) updateTriggerConsensus() {
+	// calculate the mode of the trigger configs to determine consensus
+
+	// 1. Make list of node configs that haven't expired
+	triggers := h.triggersConfig.triggersConfigMap
+
+	// 2. Make list of hashes of each node's config.
+	// 3. Count the number of times each hash appears.
+
+	triggersHashmap := make(map[string]int)
+	triggersHashmapByNodeId := make(map[string]string)
+
+	for index, triggerConfig := range triggers {
```

Comment:
```suggestion
	for nodeID, triggerConfig := range triggers {
```
Is a more descriptive name than index